### ISSUE TITLE
error: incompatible types in assignment when building with Docker/go

### ISSUE THREAD

**Issue by alexellis** [opencontainers/runc#367]
While building docker tagged branch v1.8.1 on following configuration:

uname -a
Linux 4.1.10-2-ARCH #1 SMP PREEMPT Sat Oct 10 08:33:21 MDT 2015 armv7l GNU/Linux

$ go version
go version go1.4.2 gccgo (GCC) 5.2.0 linux/arm

*During the build I get this error:*

```
vendor/src/github.com/opencontainers/runc/libcontainer/netlink/netlink_linux.go: 1260:26: error: incompatible types in assignment (cannot use type uint8 as type int8)

ifr.IfruHwaddr.Data[i] = ifrDataByte(hw[i])

^
```

Please could you advise?

**Comment by hqhq** [comment]
There is a similar issue: https://github.com/docker/docker/issues/16701
I don't have a box to reproduce this, maybe ping @brahmaroutu since there is a similar issue on ppc64le with gccgo.

**Comment by hqhq** [comment]
netlink has been refactored, but I think the uint8 and int8 usage is not changed. So it should be existed on the latest code.

**Comment by alexellis** [comment]
I managed to get around this error by using go 1.4.2 rather than gcc-go 1.4.2... binaries available for ARM - http://dave.cheney.net/unofficial-arm-tarballs

The issues appears to be with gcc-go then. 

Alex

**Comment by brahmaroutu** [comment]
@hqhq is right, this was a point in time failure, fixed in the latest code. GCCGO issue.

**Comment by hqhq** [comment]
Great, since the latest gccgo has fixed it, and @alexellis already workarounded it, I think we can close this issue.

**Comment by alexellis** [comment]
Thanks everyone.

**Comment by jordimassaguerpla** [comment]
Can you give us more info on the fix in both docker and gcc-go? I am having this same issue when packaging docker with gcc-go ... thanks

**Comment by jordimassaguerpla** [comment]
is this one

https://github.com/gcc-mirror/gcc/commit/e487326d35e0bd2acf2920ab402a028571c99131

?

**Comment by laboger** [comment]
Hi,

Here is the original golang issue on this problem https://github.com/golang/go/issues/11469.

As a result a fix was needed in gccgo and in Docker.  Unfortunately the gccgo fix did not make it into gcc 5.2 but shortly after.  The fix went into gcc trunk as well.  Building gccgo from a current snapshot in the gcc 5 branch or from gcc trunk will have the correct gccgo fix.  It will be in the next release from the gcc 5 branch after 5.2 (5.2.1 or 5.3 not sure which) and the gcc 6 release.

Sorry for the inconvenience.
